### PR TITLE
Delete datacatalog.proto

### PR DESCRIPTION
**Pull request by sonjaer**
# TL;DR
Delete this file as it seems it is [not used](https://github.com/lyft/datacatalog/blob/master/Makefile#L20.). 
Link to [datacatalog repo proto folder](https://github.com/lyft/datacatalog/tree/master/protos/idl)

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by kumare3** [comment]
I think this is fair to delete, you might have to recreate the request @sonjaer

**Comment by EngHabu** [comment]
Still needed now that we have moved the IDL to flyteidl?

**Comment by kumare3** [comment]
@tnsetting beat you to it. closing the PR
